### PR TITLE
Refactor all op->emitOpError with notifyMatchFailure

### PR DESCRIPTION
**Pull request by Jerry-Ge**
This is an refactoring patch which comes from the discussion here: [link](https://github.com/tensorflow/tensorflow/pull/56926#discussion_r932678395). 

Signed-off-by: Jerry Ge <jerry.ge@arm.com>

### CODE REVIEW

**Comment by Jerry-Ge** [comment]
@jpienaar Hi Jacques, this is a follow up PR of #56926 which had the following conversation [Link](https://github.com/tensorflow/tensorflow/pull/56926#discussion_r932678395). 

It's quite repetitive, could you help review this when you are available? Tks.

**Comment by Jerry-Ge** [comment]
@eric-k256 add Eric in CC.

File: tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc [hunk 1]
```diff
@@ -135,7 +135,8 @@ llvm::Optional<Value> convertPackOp(PatternRewriter& rewriter, Operation* op,
 
   // Check for ranked tensor type.
   if (!result_type) {
-    op->emitOpError("PackOp: result type not ranked tensor");
+    (void)rewriter.notifyMatchFailure(op,
+        "PackOp: result type not ranked tensor");
```
Comment (jpienaar) [inline tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc:139]: Nit: drop "PackOp:" , the rest follows the error conventions and the failure will point to op directly so there shouldn't be need to repeat op type.

File: tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc [hunk 2]
```diff
@@ -145,13 +146,15 @@ llvm::Optional<Value> convertPackOp(PatternRewriter& rewriter, Operation* op,
...
+    (void)rewriter.notifyMatchFailure(op,
+        "PackOp: input type not ranked tensor");
     return llvm::None;
   }
 
   input_type = inputs[0].getType().dyn_cast<RankedTensorType>();
   if (!input_type) {
-    op->emitOpError("Input 0 type not ranked tensor.");
+    (void)rewriter.notifyMatchFailure(op,
+        "Input 0 type not ranked tensor.");
```
Comment (jpienaar) [inline tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc:157]: input for consistency with others (sentence fragment, starting lower case no trailing punctuation)

**Review by jpienaar — APPROVED** [review]
Looks good with general change to make error messages consistent, thanks! (I'm OOO but this LGTM, @rsuderman could redirect while I'm out)

**Comment by gbaned** [comment]
Hi @Jerry-Ge Can you please check @jpienaar's comments and keep us posted ? Thank you!

**Comment by Jerry-Ge** [comment]
> gbaned

Thanks for the follow-up! We're currently on holiday and working on this. Will be back to you very quickly.

**Comment by Jerry-Ge** [comment]
@gbaned @jpienaar @rsuderman , Just pushed another commit with the updated fix based on @jpienaar's review. Pls let me know what else need to be updated. Tks

**Comment by Jerry-Ge** [comment]
The ARM CI test failed. Seems due to flaky.

**Review by jpienaar — APPROVED** [review]
Let me try and see if it goes through this time.

**Pull request merged**